src/peft/helpers.py

```diff
@@ -146,3 +148,20 @@ def check_if_peft_model(model_name_or_path: str) -> bool:
         is_peft_model = False
 
     return is_peft_model
+
+
+@contextmanager
+def set_adapter_scale(model, alpha):
+    # 1. TODO: Check whether scaling is prohibited on model
```

Comment:
This has yet to be implemented. E.g. you could check that the `original_scaling` is not empty after the loop exits. If it's empty, it probably means that the user passed an incompatible model and an error message with a good error message should be raised. There should be a test for that.

Reply:
I have made changes to the code to test this. Could you give a quick skim and let me know you had a similar vision?

Reply:
Added an initial doctring to the function. I think we need to iterate on the docstring and make it more verbose.